src/common/middlewares/ErrorHandler.ts

```diff
@@ -1,24 +1,51 @@
 import { Request, Response, NextFunction, ErrorRequestHandler } from 'express';
 import { inject, injectable } from 'tsyringe';
 import { StatusCodes } from 'http-status-codes';
+import { InputValidationError } from 'openapi-validator-middleware';
 import { Services } from '../constants';
 import { ILogger } from '../interfaces';
+import { HttpError } from '../exceptions/httpError';
 
-export type HttpError = Error & { status?: number };
 @injectable()
 export class ErrorHandler {
   public constructor(@inject(Services.LOGGER) private readonly logger: ILogger) {}
 
   public getErrorHandlerMiddleware(): ErrorRequestHandler {
     return (
-      err: HttpError,
+      err: Error,
       req: Request,
       res: Response,
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       next: NextFunction
     ): void => {
-      this.logger.log('error', `${req.method} request to ${req.originalUrl}  has failed with error: ${err.message}`);
-      res.status(err.status ?? StatusCodes.INTERNAL_SERVER_ERROR).json({ message: err.message });
+      let status: number;
+      let body: unknown;
+      if (err instanceof InputValidationError) {
+        status = StatusCodes.BAD_REQUEST;
+        body = {
+          message: 'request validation failed',
+          validationErrors: err.errors,
+        };
+      } else if (err instanceof HttpError) {
+        status = err.status;
```

Comment:
why not to log it?

Reply:
this error is probably due to user error that passed validation and shouldn't be log (duplicate error and such).
 in case its error that should be logged it is supposed to be logged when thrown.
 
 this handler is used in some of our outer services and wasn't modified.